File: csharp/src/AutomationRoot/Automation.Testing/Containers/StudentTests.cs
using Automation.Core.Components;
using Automation.Framework.RestApi.Pages;
using Automation.Testing.Cases;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;

namespace Automation.Testing.Containers
{
    [TestClass]
    public class StudentTests
    {
        [DataTestMethod]
        [DataRow("" +
            "{" +
            "'driver':'CHROME'," +
            "'keyword':" +
            "'Alexander'," +
            "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
            "'fluent':'Automation.Core.Components.FluentUi'," +
            "'students':'Automation.Framework.Ui.Pages.StudentsUi'" +
            "}")]
        [DataRow("" +
            "{" +
            "'driver':'HTTP'," +
            "'keyword':" +
            "'Alexander'," +
            "'application':'https://gravitymvctestapplication.azurewebsites.net'," +
            "'fluent':'Automation.Core.Components.FluentRest'," +
            "'students':'Automation.Framework.RestApi.Pages.StudentsRest'" +
            "}")]
        public void SearchStudentTest(string testParams)
        {
            // generate  test-parameters
            var parameters = JsonConvert.DeserializeObject<Dictionary<string, object>>(testParams);

            // execute with parameters
            var actual = new SearchStudents().WithTestParams(parameters).Execute().Actual;

            // assert results
            Assert.IsTrue(actual);
        }

        [DataTestMethod]
        [DataRow("" +
            "{" +
            "'driver':'CHROME'," +
            "'keyword':" +
            "'Alexander'," +
            "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
            "'fluent':'Automation.Core.Components.FluentUi'," +
            "'students':'Automation.Framework.Ui.Pages.StudentsUi'" +
            "}")]
        [DataRow("" +
            "{" +
            "'driver':'HTTP'," +
            "'keyword':" +
            "'Alexander'," +
            "'application':'https://gravitymvctestapplication.azurewebsites.net'," +
            "'fluent':'Automation.Core.Components.FluentRest'," +
            "'students':'Automation.Framework.RestApi.Pages.StudentsRest'" +
            "}")]
        public void StudentDetailsTest(string testParams)
        {
            // generate  test-parameters
            var parameters = JsonConvert.DeserializeObject<Dictionary<string, object>>(testParams);

            // execute with parameters
            var actual = new StudentDetails().WithTestParams(parameters).Execute().Actual;

            // assert results
            Assert.IsTrue(actual);
        }

        [DataTestMethod]
        //[DataRow("" +
        //    "{" +
        //    "'driver':'CHROME'," +
        //    "'firstName':'csharp'," +
        //    "'lastName':'student'," +
        //    "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
        //    "'fluent':'Automation.Core.Components.FluentUi'," +
        //    "'students':'Automation.Framework.Ui.Pages.StudentsUi'" +
        //    "}")]
        [DataRow("" +
            "{" +
            "'driver':'HTTP'," +
            "'firstName':'csharp'," +
            "'lastName':'student'," +
            "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
            "'fluent':'Automation.Core.Components.FluentRest'," +
            "'students':'Automation.Framework.RestApi.Pages.StudentsRest'" +
            "}")]
        public void CreateStudentUiTest(string testParams)
        {
            // generate  test-parameters
            var parameters = JsonConvert.DeserializeObject<Dictionary<string, object>>(testParams);

            // execute with parameters
            var actual = new CreateStudent().WithTestParams(parameters).Execute().Actual;

            // assert results
            Assert.IsTrue(actual);
        }
    }
}

File: java/src/automation-root/automation-testing/src/test/java/com/automation/testing/containers/StudentTests.java
package com.automation.testing.containers;

import com.automation.testing.cases.CreateStudent;
import com.automation.testing.cases.SearchStudents;
import com.automation.testing.cases.StudentDetails;
import com.automation.testing.tools.Utilities;
import org.testng.Assert;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

import java.net.MalformedURLException;
import java.util.Map;

public class StudentTests {
    @Test(dataProvider = "search-data-provider")
    public void searchStudentTest(Map<String, Object> testParams) throws MalformedURLException {
        // execute & get actual result
        boolean actual = new SearchStudents().withTestParams(testParams).execute().getActual();

        //  assert actual
        Assert.assertTrue(actual);
    }

    @Test(dataProvider = "details-data-provider")
    public void studentDetailsTest(Map<String, Object> testParams) throws MalformedURLException {
        // execute & get actual result
        boolean actual = new StudentDetails().withTestParams(testParams).execute().getActual();

        //  assert actual
        Assert.assertTrue(actual);
    }

    @Test(dataProvider = "create-data-provider")
    public void createStudentTest(Map<String, Object> testParams) throws MalformedURLException {
        // execute & get actual result
        boolean actual = new CreateStudent().withTestParams(testParams).execute().getActual();

        //  assert actual
        Assert.assertTrue(actual);
    }

    @DataProvider(name = "search-data-provider")
    public Object[][] searchStudentProvider() {
        return searchProvider();
    }

    @DataProvider(name = "details-data-provider")
    public Object[][] studentDetailsProvider() {
        return searchProvider();
    }

    @DataProvider(name = "create-data-provider")
    public Object[][] createStudentProvider() {
        // data to pass as test parameters
        String json = "{" +
                "'driver':'CHROME'," +
                "'driverBinaries':'D:\\\\automation-env\\\\web-drivers'," +
                "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
                "'firstName':'java'," +
                "'lastName':'student'" +
                "}";

        // return data-provider
        return Utilities.dataProviderFactory(json);
    }

    private Object[][] searchProvider(){
        // data to pass as test parameters
        String json = "[" +
                "{" +
                "'driver':'CHROME'," +
                "'driverBinaries':'D:\\\\automation-env\\\\web-drivers'," +
                "'application':'https://gravitymvctestapplication.azurewebsites.net/Student'," +
                "'keyword':'Alexander'," +
                "'fluent':'com.automation.core.components.FluentUi'," +
                "'students':'com.automation.framework.ui.pages.StudentsUi'" +
                "}," +
                "{" +
                "'driver':'HTTP'," +
                "'application':'https://gravitymvctestapplication.azurewebsites.net'," +
                "'keyword':'Alexander'," +
                "'fluent':'com.automation.core.components.FluentRest'," +
                "'students':'com.automation.framework.rest.pages.StudentsRest'" +
                "}" +
                "]";

        // return data-provider
        return Utilities.dataProviderFactory(json);
    }
}